### PR TITLE
Create 'shared-web' engine, move parts of web into it to share with cosmetics-web

### PR DESCRIPTION
**Pull request by DWRendell**

### CODE REVIEW

File: cosmetics-web/deploy.sh [hunk 1]
```diff
@@ -11,6 +11,8 @@ set -ex
 # CF_PASSWORD: cloudfoundry password
 # SPACE: the space to which you want to deploy
 
+cp ./shared-web ./cosmetics-web/vendor/shared-web
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply cosmetics-web/deploy.sh]: Oh good point, adding -R

File: shared-web/.gitignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.bundle/
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply shared-web/.gitignore]: Yes, done.

File: shared-web/MIT-LICENSE [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Copyright 2018 David Rendell
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply shared-web/MIT-LICENSE]: Removed, I blame rails/yarn

File: shared-web/shared-web.gemspec [hunk 4]
```diff
@@ -0,0 +1,22 @@
+$:.push File.expand_path("lib", __dir__)
+
+# Maintain your gem's version:
+require "shared/web/version"
+
+# Describe your gem and declare its dependencies:
+Gem::Specification.new do |s|
+  s.name        = "shared-web"
+  s.version     = Shared::Web::VERSION
+  s.authors     = ["David Rendell"]
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply shared-web/shared-web.gemspec]: Sure. What shall I put for the email? The Softwire DL?

File: shared-web/test/dummy/.ruby-version [hunk 5]
```diff
@@ -0,0 +1 @@
+2.4.4
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply shared-web/test/dummy/.ruby-version]: Probably not, will remove all this.

File: cosmetics-web/deploy.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (broder) [reply cosmetics-web/deploy.sh]: We're using `-a` in e.g. https://github.com/UKGovernmentBEIS/beis-mspsds/blob/master/worker/deploy.sh - might be better to be consistent.

File: shared-web/shared-web.gemspec (hunk 4 above)
[the remark this answers is not in the document]
Comment (broder) [reply shared-web/shared-web.gemspec]: We probably don't need to define it?

File: cosmetics-web/Gemfile [hunk 6]
```diff
@@ -42,5 +44,9 @@ group :development do
 
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
+
+  # Shared web will be accessed from the root directory in development, and be copied to the vendors directory for deployment
+  shared_web_path = '../shared-web'
```
[the remark this answers is not in the document]
Comment (DWRendell) [reply cosmetics-web/Gemfile]: I haven't managed to get debugging working, on this branch or on master, so not sure right now. Might be worth going over this when we're in the same place.

[the remark this answers is not in the document]
Comment (DWRendell) [reply cosmetics-web/Gemfile]: I've managed to get the debugging working, and it seems to still work. I haven't tried adding breakpoints in the shared code as I can't think of anything in there at the moment that's sensible to add one to, but I think it should work.

File: cosmetics-web/Gemfile.lock [hunk 7]
```diff
@@ -9,6 +9,12 @@ GIT
       nokogiri (>= 1.6.6)
       rubyzip (>= 1.2.1)
 
+PATH
+  remote: ../shared-web
```
Comment (broder) [inline cosmetics-web/Gemfile.lock]: I'm a little bit concerned that this isn't the `shared-web` path for production. The cloud foundry deployment runs `bundle install --deployment` which just looks at the `Gemfile.lock` - could that cause issues for resolving the `shared-web` code?

**Pull request merged**